src/targets/maven.ts

```diff
@@ -187,31 +187,117 @@ export class MavenTarget extends BaseTarget {
    * @param revision Git commit SHA to be published.
    */
   public async publish(_version: string, revison: string): Promise<void> {
-    await this.createUserGradlePropsFile();
-    await this.upload(revison);
+    await withTempDir(async dir => {
+      const snapshotPath = await this.makeGradlePropsSnapshot(dir);
```

Comment:
I mean you've come all this way, it is trivial to convert this into something like `withTempDir` that wraps around the actions you want to run safely.